packages/core/core/prompt/attachments-to-parts.ts

```diff
@@ -0,0 +1,74 @@
+import { Attachment } from '@ai-sdk/ui-utils';
+import { ImagePart, TextPart } from './content-part';
+import { convertDataContentToUint8Array } from './data-content';
+import { getErrorMessage } from '@ai-sdk/provider-utils';
+
+type ContentPart = TextPart | ImagePart;
+
+/**
+ * Converts a data URL of type text/* to a text string.
+ */
+export function dataUrlToText(dataUrl: string): string {
+  return Buffer.from(dataUrl, 'base64').toString('utf-8');
+}
+
+/**
+ * Checks if a string is a Data URL.
+ */
+function isDataURL(url: string): boolean {
+  return new URL(url).protocol === 'data:';
```

Comment:
try...catch needed if url parsing fails imo:

```
try {
  return new URL(url).protocol === 'data:';
} catch (ignored) {
  return false
  }

Reply:
this has be removed, so no longer relevant